esx_service/vmodl/VsphereContainerServiceImpl.py

```diff
@@ -0,0 +1,269 @@
+
+        error_info = auth_api._tenant_vm_rm(tenant.name, vms)
+        if error_info:
+            logging.error("Failed to remove VMs: %s from tenant: %s", vms, tenant.name)
+            raise vim.fault.VcsFault(msg=error_info.msg)
+
+        logging.info("Succssfully removed VMs: %s from tenant: %s", vms, tenant.name)
+    
+    def ReplaceVMs(self, tenant, vms):
+        if len(vms) == 0:
+            logging.error("Replace VMs: the VM list is empty")
+            raise vmodl.fault.InvalidArgument("VM list is empty")
+
+        logging.info("Replacing VMs for tenant: %s", tenant.name)
+        logging.info("Existing VMs: %s", tenant.vms)
+        logging.info("New VMs: %s", vms)
+
+        error_info = auth_api._tenant_vm_replace(tenant.name, vms)
+        if error_info:
+            logging.error("Failed to replace VMs for tenant: %s", tenant.name)
+            raise vim.fault.VcsFault(msg=error_info.msg)
+
+        logging.info("Succssfully replaced VMs for tenant: %s", tenant.name)
+
+    def AddPrivilege(self, tenant, privilege, default_datastore=False):
+        logging.info("Adding privilege: %s to tenant: %s, default_datastore=%s",
+            privilege, tenant.name, default_datastore)
+
+        error_info = auth_api._tenant_access_add(name=tenant.name,
+                                                 datastore=privilege.datastore,
+                                                 allow_create=privilege.allow_create,
+                                                 volume_maxsize_in_MB=privilege.volume_max_size,
+                                                 volume_totalsize_in_MB=privilege.volume_total_size,
+                                                 default_datastore=default_datastore)
+        if error_info:
+            logging.error("Failed to add privilege to tenant: %s", tenant.name)
+            raise vim.fault.VcsFault(msg=error_info.msg)
+
+        logging.info("Succssfully added privilege to tenant: %s", tenant.name)
+
+    def UpdatePrivilege(self, tenant, datastore, allow_create, volume_max_size, volume_total_size):
+        logging.info("Updating privilege (datastore=%s) for tenant: %s", datastore, tenant.name)
+
+        error_info = auth_api._tenant_access_set(name=tenant.name,
+                                                 datastore=datastore,
+                                                 allow_create=allow_create,
+                                                 volume_maxsize_in_MB=volume_max_size,
+                                                 volume_totalsize_in_MB=volume_total_size)
+        if error_info:
+            if error_info.code == ErrorCode.PRIVILEGE_NOT_FOUND:
+                logging.error(error_info.msg)
+                raise vim.fault.VcsFault(msg=error_info.msg)
+            else:
+                logging.error("Failed to update privilege for tenant: %s", tenant.name)
+                raise vim.fault.VcsFault(msg=error_info.msg)
+
```

Comment:
In both if and else parts the same fault is being raised can "vim.fault.VcsFault(msg=error_info.msg)" be placed outside of the if-else block.